### PR TITLE
Feature/99 open cms help in overlay

### PR DESCRIPTION
**Pull request by pixeljunkie**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#99
| License       | MIT

Includes styling the help module and a handling iframe mode or standalone mode. Reworked the module but did not replace all of the deprecated calls.

### CODE REVIEW

File: src/CoreBundle/private/modules/MTHeader/views/standard.view.php [hunk 1]
```diff
@@ -253,13 +251,13 @@ class="nav-link dropdown-toggle <?= $userButtonStyle ?>"
                             </span>
                         </a>
                         <div class="dropdown-menu dropdown-menu-right">
-                            <a class="dropdown-item" href="<?= PATH_CMS_CONTROLLER ?>?pagedef=tableeditor&tableid=<?= $data['iTableIDCMSUser'] ?>&id=<?= $data['oUser']->id ?>&<?= urlencode('module_fnc[contentmodule]') ?>">
+                            <a class="dropdown-item" href="<?= PATH_CMS_CONTROLLER; ?>?pagedef=tableeditor&tableid=<?= $data['iTableIDCMSUser']; ?>&id=<?= $data['oUser']->id; ?>&<?= urlencode('module_fnc[contentmodule]'); ?>">
```
Comment (umulmrum) [inline src/CoreBundle/private/modules/MTHeader/views/standard.view.php]: Those semicolons aren't required and are omitted by most projects, and I would omit them, too (in fact I removed some in the last few days).

File: src/CoreBundle/private/modules/CMSModuleHelp/CMSModuleHelp.class.php [hunk 2]
```diff
@@ -25,23 +29,37 @@ class CMSModuleHelp extends TCMSModelBase
...
-    {
-    }
-
     public function &Execute()
     {
         $this->data = parent::Execute();
 
+        $translator = $this->getTranslator();
+
+        $this->data['translator'] = $translator;
```
Comment (umulmrum) [inline src/CoreBundle/private/modules/CMSModuleHelp/CMSModuleHelp.class.php]: I'd advocate against passing service dependencies to views as it increases coupling between module and view.

File: src/CoreBundle/private/modules/MTHeader/views/standard.view.php (hunk 1 above)
Comment (pixeljunkie) [reply src/CoreBundle/private/modules/MTHeader/views/standard.view.php]: CS-Fixer did this. Without changing the code style config i don`t get rid of them.

File: src/CoreBundle/private/modules/CMSModuleHelp/CMSModuleHelp.class.php (hunk 2 above)
Comment (pixeljunkie) [reply src/CoreBundle/private/modules/CMSModuleHelp/CMSModuleHelp.class.php]: Right. Now setting the translated title in the module instead.

File: src/CoreBundle/private/modules/MTHeader/views/standard.view.php (hunk 1 above)
Comment (umulmrum) [reply src/CoreBundle/private/modules/MTHeader/views/standard.view.php]: Looks like I did it wrong :-) - OK, I don't like the semicolons in this case, but we stick with the CS-Fixer defaults.

**Comment by umulmrum** [comment]
OK, now only the question "target blank vs. overlay" is open.

**Comment by pixeljunkie** [comment]
Blank vs overlay? I managed to allow both. default = overlay. CTRL click = blank. I think this is the best solution. The links in the help are targeted with top, so the links work inside the overlay.

**Comment by umulmrum** [comment]
OK rechecked - looks good to me.

But there are merge conflicts. Could you resolve?

**Comment by pixeljunkie** [comment]
resolved merge conflict

**Comment by umulmrum** [comment]
Thanks @pixeljunkie

**Pull request merged**